src/coll/options.rs

```diff
@@ -49,6 +49,20 @@ pub enum ReturnDocument {
     Before,
 }
 
+impl<'de> Deserialize<'de> for ReturnDocument {
```

Comment:
I think this could be achieved via the derived `Deserialize` implementation and `#[serde(rename_all="lowercase")]`

Reply:
This is a little tedious, but it's possible to specify multiple alternate names with [`alias`](https://serde.rs/field-attrs.html#alias). I don't think we'd want to use it for names longer than one word, but for this case it might be a reasonable choice.